File: src/routes/url.route.ts
import { Router } from 'express';
import { UrlController } from '../controllers/url.controller';
import { UrlService } from '../services/url.service';

const router = Router();

const urlService = new UrlService();
const urlController = new UrlController(urlService);

router.post('/shorten-url', (req, res) => urlController.shortenUrl(req, res));

router.get('/:id', (req, res) => urlController.redirectUrl(req, res));

export default router;

File: src/routes/url_route.go
package routes

import (
	"gotcha/src/controllers"
	"gotcha/src/db"
	"gotcha/src/services"

	"github.com/gin-gonic/gin"
)

func SetupRoutes(router *gin.Engine) {
	database := db.NewDatabase("mongodb://localhost:27017")
	if err := database.Connect(); err != nil {
		panic(err)
	}

	// Initialize services
	urlService, err := services.NewURLService(database)
	if err != nil {
		panic(err)
	}

	// Initialize controllers
	urlController := controllers.NewURLController(urlService)

	// URL routes
	router.POST("/shorten-url", urlController.CreateShortURL)
	router.GET("/:short", urlController.RedirectToOriginal)
}
